### PR TITLE
code_warntype docs: more neutral reference to @code_warntype

### LINKED ISSUE #51358: Documentation for code_warntype tells to look for @code_warntype but that one is useless
This is a documentaion issue. Consider the last line of the `help` message for `code_warntype`:

```
help?> code_warntype
search: code_warntype @code_warntype

  code_warntype([io::IO], f, types; debuginfo=:default)

  Prints lowered and type-inferred ASTs for the methods matching
  the given generic function and type signature to io which
  defaults to stdout. The ASTs are annotated in such a way as to
  cause "non-leaf" types which may be problematic for performance
  to be emphasized (if color is available, displayed in red).
  This serves as a warning of potential type instability.

  Not all non-leaf types are particularly problematic for
  performance, and the performance characteristics of a
  particular type is an implementation detail of the compiler.
  code_warntype will err on the side of coloring types red if
  they might be a performance concern, so some types may be
  colored red even if they do not impact performance. Small
  unions of concrete types are usually not a concern, so these
  are highlighted in yellow.

  Keyword argument debuginfo may be one of :source or :none
  (default), to specify the verbosity of code comments.

  See @code_warntype for more information.
```
So, I'm ready to dive deeper and fire `help` for `@code_warntype`:
```
help?> @code_warntype
  @code_warntype

  Evaluates the arguments to the function or macro call,
  determines their types, and calls code_warntype on the
  resulting expression.
```
And that is it.

So, either the last line of `help` for `code_warntype` should be removed, or `help` for `@code_warntype` should be expanded.

```
julia> versioninfo()
Julia Version 1.9.3
Commit bed2cd540a1 (2023-08-24 14:43 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 16 × AMD Ryzen 9 6900HX with Radeon Graphics
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.6 (ORCJIT, znver3)
  Threads: 1 on 16 virtual cores
```

Julia installed with `snap`.

### PR DESCRIPTION
**Pull request by ulysses4ever**
fix #51358

### CODE REVIEW

**Comment by christiangnrd** [comment]
I don’t think the old reference should be completely removed as when accessing it from the docs it links to https://docs.julialang.org/en/v1/manual/performance-tips/#man-code-warntype which actually does have more information.

Maybe rewording it to make it clear that it’s linking to a section of the Performance Tips page in the manual would be more helpful.

**Comment by christiangnrd** [comment]
Maybe something like

`See the @code_warntype section in the Performance Tips page of the manual for more information.’

**Comment by ulysses4ever** [comment]
I updated it as suggested.

**Comment by christiangnrd** [comment]
You want to modify line 73 without removing line 75.

**Comment by ulysses4ever** [comment]
My bad. Another attempt is in.

File: stdlib/InteractiveUtils/src/codeview.jl [hunk 1]
```diff
@@ -70,7 +70,7 @@ Small unions of concrete types are usually not a concern, so these are highlight
 
 Keyword argument `debuginfo` may be one of `:source` or `:none` (default), to specify the verbosity of code comments.
 
-See [`@code_warntype`](@ref man-code-warntype) for more information.
+See the @code_warntype section in the Performance Tips page of the manual for more information.
```
Comment (KristofferC) [inline stdlib/InteractiveUtils/src/codeview.jl:73]: Now this doesn't link to the manual anymore in the docs, or? (Also macro should be code quoted)

Comment (ulysses4ever) [reply stdlib/InteractiveUtils/src/codeview.jl]: Right. How about now?

**Review by fingolfin — APPROVED** [review]
Looks good to me, thanks!
